wsagent/che-core-api-debug/src/main/java/org/eclipse/che/api/debugger/server/DebuggerService.java

```diff
@@ -165,17 +174,36 @@ public void deleteBreakpoint(
   }
 
   @GET
-  @Path("{id}/dump")
+  @Path("{id}/stackframedump")
   @Produces(MediaType.APPLICATION_JSON)
-  public StackFrameDumpDto getStackFrameDump(@PathParam("id") String sessionId)
+  public StackFrameDumpDto getStackFrameDump(
+      @PathParam("id") String sessionId,
+      @QueryParam("thread") @DefaultValue("-1") long threadId,
+      @QueryParam("frame") @DefaultValue("-1") int frameIndex)
+      throws DebuggerException {
+    if (threadId == -1) {
+      return asDto(debuggerManager.getDebugger(sessionId).dumpStackFrame());
```

Comment:
`getDebugger(String sessionId)` You have deprecated this method, why are you still using it?